services/worker-manager/src/providers/aws.js

```diff
@@ -107,6 +89,25 @@ class AwsProvider extends Provider {
         ts.ResourceType === 'instance' ? instanceTags.concat(ts.Tags) : otherTagSpecs.push(ts)
       );
 
+      const userData = Buffer.from(JSON.stringify({
+        rootUrl: this.rootUrl,
+        workerPoolId,
+        providerId: this.providerId,
+        workerGroup: this.providerId,
+        workerConfig: config.launchConfig.workerConfig || {},
+      }));
+      // https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-metadata.html#instancedata-add-user-data
+      // The raw data should be 16KB maximum
+      if (userData.length > 16384) {
+        return await workerPool.reportError({
+          kind: 'creation-error',
+          title: 'User Data is too long',
+          description: 'Try a shorter workerPoolId and/or a shorter rootUrl',
```

Comment:
Maybe better advice here is "try removing some workerConfiguration and consider putting it in a secret"?